src/k8s/pkg/k8sd/app/hooks_post_refresh.go

```diff
@@ -110,3 +119,48 @@ func (a *App) performPostUpgrade(ctx context.Context, s state.State) error {
 
 	return nil
 }
+
+func (a *App) updateNodeIPAddresses(ctx context.Context, s state.State) error {
```

Comment:
is it REALLY a good idea to expose the kubelet on EVERY nic on the system?

Reply:
It probably isn't. In a cloud scenario, where you'd have multiple network planes, Kubernetes internals is the sort of thing you'd limit to only management or internal network planes. Is there a way to specify the NICs to use for this in `k8s-snap`?

Ihis looks similar to the workflow in hooks_join / hooks_bootstrap, so, I guess @addyess question would also apply there. In any case, could we create an util function for this and reusing it in all 3 cases?

Reply:
I believe we do not expose/use every nic/interface. The cluster member derives the address(and through this the interface) from the `--address` flag provided to bootstrap or join commands.

The IP list here is extracted from the interface that the member IP address belongs to. The steps are basically the same but I'm not sure if we can implement a common util function across all 3 cases. (Most of these changes below are common steps in join/bootstrap hooks and stays at the upper level function, and is passed down the the `setup_kubelet`.)